cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -143,6 +143,9 @@ type APIServer struct {
 
 	// TimeoutForControlPlane controls the timeout that we use for API server to appear
 	TimeoutForControlPlane *metav1.Duration
+
+	// DNSPolicy sets a DNS policy for the API Server.
+	DNSPolicy v1.DNSPolicy
```

Comment:
hi, i don't think we have this in documentation yet, but after the betas kubeadm API changes fall under a strict review process where we discuss them first.

a couple of points:
- v1beta1 and v1beta2 are locked at this point and cannot be changed.
- we had PRs before to modify the default policy to use, but not feature-requests yet to make this configurable.
i suggest you log a ticket in k/kubeadm about this and ping me on it, so that it can see discussion. also we meet every wednesday. https://github.com/kubernetes/community/tree/master/sig-cluster-lifecycle (kubeadm office hours)
- there is an issue in k/kubeadm that tracks support for similar customizations, so not sure whether this will be accepted to be part of the API, but maybe it will remain as patchable per-node / pod. https://github.com/kubernetes/kubeadm/issues/1379 

/hold

Reply:
> v1beta1 and v1beta2 are locked at this point and cannot be changed.
Wow, I wasn't aware of that. 

> we had PRs before to modify the default policy to use, but TMK no feature-requests yet to make this configurable.

You had a PR that harcoded the dns policy to be `ClusterFirstWithHostNet` but that was an opinionated decision, that is why we believe that enabling this property would be beneficial.

I'll join the meeting today and create an issue in k/kubeadm.

Reply:
@neolit123 I created the issue https://github.com/kubernetes/kubeadm/issues/1637